### PR TITLE
Catch Ruby exceptions during processing

### LINKED ISSUE #2: Exceptions cause a filterworker to die
An exception (NameError, NoMethodError, etc) will cause the filterworker to die off leaving basically nothing reading from the input-to-filter queue.

Should be  easy to fix, and we should include a test case for this as well.

---

On error, the filter worker should continue like normal. Perhaps we should add a tag _rubyexception?

---

Example

```
% bin/logstash -e 'filter { ruby { code => "whatever" } }'
hello world
Using milestone 1 filter plugin 'ruby'. This plugin should work, but would benefit from use by folks like you. Please let us know if you find bugs or have suggestions on how to improve this plugin.  For more information on plugin milestones, see http://logstash.net/docs/2.0.0.dev/plugin-milestones {:level=>:warn}

Exception in filterworker {"exception"=>#<NameError: undefined local variable or method `whatever' for <LogStash::Filters::Ruby code=>"whatever">:LogStash::Filters::Ruby>, "backtrace"=>["(ruby filter code):1:in `register'", "org/jruby/RubyProc.java:271:in `call'", "/Users/jls/projects/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-ruby-0.1.2/lib/logstash/filters/ruby.rb:38:in `filter'", "(eval):50:in `initialize'", "org/jruby/RubyArray.java:1613:in `each'", "org/jruby/RubyEnumerable.java:805:in `flat_map'", "(eval):46:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/Users/jls/projects/logstash/lib/logstash/pipeline.rb:272:in `filter'", "/Users/jls/projects/logstash/lib/logstash/pipeline.rb:206:in `filterworker'", "/Users/jls/projects/logstash/lib/logstash/pipeline.rb:150:in `start_filters'"], :level=>:error}
```

### PR DESCRIPTION
**Pull request by netoneko**
Fixes #2

### CODE REVIEW

File: lib/logstash/filters/ruby.rb [hunk 1]
```diff
@@ -34,7 +34,12 @@ def register
   def filter(event)
     return unless filter?(event)
 
-    @codeblock.call(event)
+    begin
+      @codeblock.call(event)
+    rescue Exception => e
```
Comment (jsvd) [inline lib/logstash/filters/ruby.rb]: if there's an exception, `filter_matched` shouldn't be called. here's a failing spec:

```ruby
  describe "a rubyexception should not call filter_matched on the event" do

    config <<-CONFIG
      filter {
        ruby {
          code => "raise 'You shall not pass'"
          add_tag => ["ok"]
        }
      }
    CONFIG

    sample("message" => "hello world") do
      insist { subject["tags"] } == ["_rubyexception"]
    end
  end
```

I'd suggest moving filter_matched(event) to:

```ruby
begin
  @codeblock.call(event)
rescue # ..
  # ...
else # called if no exceptions were raised
  filter_matched(event)
end
```

Comment (netoneko) [inline lib/logstash/filters/ruby.rb]: This will work too:

```
begin
  @codeblock.call(event)
  filter_matched(event)
rescue # ..
  # ...
end
```

**Comment by netoneko** [comment]
I've pushed another commit that addresses your comment.

**Comment by jsvd** [comment]
would you mind doing step 2 of the [Contributing Guide](https://github.com/elastic/logstash/blob/master/CONTRIBUTING.md#contribution-steps)?

**Comment by netoneko** [comment]
@jsvd done.

**Comment by jsvd** [comment]
LGTM

**Pull request closed without merging**